### PR TITLE
Migrate Network panel to new NetworkRequestsCache

### PR DESCRIPTION
**Pull request by bvaughn**
- [x] Update Replay client API for how network requests and responses are fetched.
- [x] Removed all Redux state/actions in favor of Suspense caches for fetching network data.
- [x] Changed `networkRequestsCache` response format slightly so that it was easier to work with for the Network table.

I didn't rewrite the view files here, just updated the `partialRequestsToCompleteSummaries` to map from the Suspense cache data format to the intermediate `RequestSummary` format. If we had more time and energy, I'd like to see us rewrite this legacy code, but...

### CODE REVIEW

**Review by markerikson — APPROVED** [review]

File: packages/replay-next/src/suspense/NetworkRequestsCache.ts [hunk 1]
```diff
@@ -1,50 +1,168 @@
...
+            time,
+          },
+          triggerPoint,
+        } as NetworkRequestsData;
+      });
+
+      data.events.forEach(({ id, event }) => {
+        const events = records[id].events;
+        switch (event.kind) {
+          case "request":
```
[the remark this answers is not in the document]
Comment (bvaughn) [reply packages/replay-next/src/suspense/NetworkRequestsCache.ts:100]: Yeah, it's pretty weird how the data streams in for this stuff– in 4 buckets, except that two are grouped together into a single handler (`RequestInfo` and `RequestEventInfo`) although they arrive separately. It makes...no sense to me.

[the remark this answers is not in the document]
Comment (bvaughn) [reply packages/replay-next/src/suspense/NetworkRequestsCache.ts:100]: Actually looking at the protocol docs again, I see:
> NOTE: There is no guarantee that request information will be available before the request event info, so all temporal combinations should be supported when processing this data.

Which makes me think I need to revisit the way I've written this block:

https://github.com/replayio/devtools/blob/f3be7cdab913d3b1f41abd7b3d55b6742e33c478/packages/replay-next/src/suspense/NetworkRequestsCache.ts#L63-L98

What a pain in the ass.

**Comment by bvaughn** [comment]
e2e tests caught a small status regression:
https://app.replay.io/recording/5569895c-cb22-41d3-8b50-495ea6a9caa5

Will fix in the morning before landing.

**Comment by jasonLaster** [comment]
Love seeing that our e2e tests are catching things

**Pull request merged**